Functionality to add accounts to a postgres database

dskeppstedt opened the pull request:
This is the first step of building the user service.
We can now add user accounts from our service to a postgres database.

dskeppstedt committed:
Add database class

dskeppstedt committed:
Add account model

dskeppstedt committed:
Add simple main program

dskeppstedt committed:
Add example file for environment

dskeppstedt committed:
Add dependencis file for python modules

dskeppstedt committed:
Add the database schema under vc

dskeppstedt committed:
Update the read me to reflect the changes to the user service.

DaRavenox reviewed (dismissed):
Code looks good. Try to use less of the colour green as background.

AndreasBrommund commented on schema/iTime.sql:
```diff
@@ -0,0 +1,98 @@
+--
```
Do we need this in git?

dskeppstedt replied:
I will have to look into this, we could store the schema somewhere. 
Not sure how yet.

dskeppstedt replied:
https://blog.codinghorror.com/get-your-database-under-version-control/

dskeppstedt replied:
I think this is the most simple solution at the moment.

dskeppstedt commented on README.md:
````diff
@@ -1,2 +1,36 @@
+export ITIME_DB="test_db"
+export ITIME_USER="test"
+export ITIME_DB_PASSWORD="password"
+```
````
source this file or put these environment variables your .profile

dskeppstedt committed:
Update README to be correct.

dskeppstedt commented on account.py:
```diff
@@ -0,0 +1,16 @@
+    
+    def save(self,database):
+        database.cursor.execute(
+        "INSERT INTO accounts (email,google_access_token,google_refresh_token) VALUES (%s,%s,%s)",
```
@HadarGreinsmark and I thinks this should be moved somewhere else as this makes the Account class tightly coupled with the current database

dskeppstedt committed:
Fix typo in env_example file

dskeppstedt committed:
Add a database model mapper for account model

I did this change to decouple the Account model from the database, if we in the future want to change the database this abstraction will help.

dskeppstedt commented:
@AndreasBrommund @DaRavenox @HadarGreinsmark please review again

DaRavenox commented on account_db_mapper.py:
```diff
@@ -0,0 +1,15 @@
+            "VALUES (%s,%s,%s)",
+            (account.email,account.access_token,account.refresh_token)
+        )
+
```
Do we really want to save the tokens in the database? They will be changed quite often and mainly only used when the user is actually requesting us to use them. Yours is probably the best solution for the time being however.

Vill vi faktiskt spara ner access_token och refresh_token i databasen då de kommer förmodligen att bytas ut rätt ofta? Jag antar dock att det är bäst att göra det så här for the time being.

dskeppstedt replied:
We need to keep the tokens in the database or store them somehow if we want to be able to access the google api. This is the server side tokens you know, not the same as the authcode.

DaRavenox replied:
I understand the what the tokens are. My point is: do we want to continuously access google API or would be better served by getting the info we want from the calendar once and then storing **that** in the database? Since the refresh tokens expire when user revokes access we might have to keep adding and removing them from the DB even during a single use session. Again, for 1.0 we might just have to accept this behaviour.

dskeppstedt replied:
This is something that can be discussed further. Will add it to Trello.

DaRavenox commented on account.py:
```diff
@@ -0,0 +1,9 @@
+        self.email = email
+        self.access_token = access_token
+        self.refresh_token = refresh_token
+    
```
Is this class just for holding the user data over until it is put in the DB? Is this class meant to be an integral part of the whole backend system?

Vad används det här till utöver att ha som en temp. hållare av datan före den är lagd i databasen? Kommer andra delar av B-E att jobba med Accounts också?

dskeppstedt replied:
Yes this is the model for Accounts in user service. This will not be shared with the other services but only within the user service.

The design pattern that I am aming for is called **M**odel-**V**iew-**C**ontroller or MVC for short.

This will make more sens further down when the user service will grow in complexity.

DaRavenox replied:
Okay. This seems reasonable then.

DaRavenox reviewed (commented):
Code looks good but see my questions before I will approve.
